django_readonly_field/compiler.py

```diff
@@ -56,7 +56,18 @@ def remove_read_only_fields(self, read_only_field_names):
         """
         fields = self.query.fields
 
-        fields[:] = (
+        new_fields = (
             field for field in fields
             if field.name not in read_only_field_names
         )
+        try:
+            fields[:] = new_fields
+        except AttributeError:
+            # When deserializing, we might get an attribute error because this
+            # list shoud be copied first :
+
+            # "AttributeError: The return type of 'local_concrete_fields'
+            # should never be mutated. If you want to manipulate this list for
+            # your own use, make a copy first."
+
+            self.query.fields = list(new_fields)
```

Comment:
Note : the fact that tests pass tends to prove that at this point, the generator has not been consumed yet, but there's actually no guarantee. Should we make it safer by recreating the generator ?